homeassistant/components/sonos/media_browser.py

```diff
@@ -64,22 +69,22 @@ def get_thumbnail_url_full(
     )
 
 
-def media_source_filter(item: BrowseMedia):
+def media_source_filter(item: BrowseMedia) -> bool:
     """Filter media sources."""
     return item.media_content_type.startswith("audio/")
 
 
 async def async_browse_media(
-    hass,
+    hass: HomeAssistant,
     speaker: SonosSpeaker,
     media: SonosMedia,
     get_browse_image_url: GetBrowseImageUrlType,
     media_content_id: str | None,
     media_content_type: str | None,
-):
+) -> BrowseMedia:
     """Browse media."""
 
-    if media_content_id is None:
+    if media_content_id is None or media_content_type is None:
```

Comment:
`media_content_id` and `media_content_type` are inclusive. I.e. either both are `str` or both `None`. Instead of the second condition, I would suggest adding an `assert` below the if clause.
```diff
    if media_content_id is None:
        return await root_payload(
            hass,
            speaker,
            media,
            get_browse_image_url,
        )
+   assert media_content_type is not None
```